src/main/java/org/wso2/carbon/inbound/salesforce/poll/LoginHelper.java

```diff
@@ -0,0 +1,185 @@
+
+        @Override
+        public void characters(char[] ch, int start, int length) {
+
+            if (reading)
+                buffer = new String(ch, start, length);
+        }
+
+        @Override
+        public void endElement(String uri, String localName, String qName) {
+
+            reading = false;
+            switch (localName) {
+                case "sessionId":
+                    sessionId = buffer;
+                    break;
+                case "serverUrl":
+                    serverUrl = buffer;
+                    break;
+                case "faultstring":
+                    faultstring = buffer;
+                    break;
+                default:
+            }
+            buffer = null;
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes attributes) {
+
+            switch (localName) {
+                case "sessionId":
+                    reading = true;
+                    break;
+                case "serverUrl":
+                    reading = true;
+                    break;
+                case "faultstring":
+                    reading = true;
+                    break;
+                default:
+            }
+        }
+    }
+
+    public static final String COMETD_REPLAY = "/cometd/";
+    public static final String COMETD_REPLAY_OLD = "/cometd/replay/";
+    static final String LOGIN_ENDPOINT = "https://login.salesforce.com";
+    private static final String ENV_END = "</soapenv:Body></soapenv:Envelope>";
+    private static final String ENV_START =
+            "<soapenv:Envelope xmlns:soapenv='http://schemas.xmlsoap.org/soap/envelope/' "
+                    + "xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' "
+                    + "xmlns:urn='urn:partner.soap.sforce.com'><soapenv:Body>";
+    // The enterprise SOAP API endpoint used for the login call
+    private static final String SERVICES_SOAP_PARTNER_ENDPOINT =
+            "/services/Soap/u/" + SalesforceDataHolderObject.getSoapApiVersion() + "/";
+
+    public static BayeuxParameters login(String username, String password) throws Exception {
+
+        return login(new URL(LOGIN_ENDPOINT), username, password);
+    }
+
+    public static BayeuxParameters login(String username, String password, BayeuxParameters params) throws Exception {
+
+        return login(new URL(LOGIN_ENDPOINT), username, password, params);
+    }
+
+    public static BayeuxParameters login(URL loginEndpoint, String username, String password) throws Exception {
+
+        return login(loginEndpoint, username, password, new BayeuxParameters() {
+            @Override
+            public String bearerToken() {
+
+                throw new IllegalStateException("Have not authenticated");
+            }
+
+            @Override
+            public URL endpoint() {
+
+                throw new IllegalStateException("Have not established replay endpoint");

```

Comment:
when you throw an exception, please give a proper meaningful error message.

Reply:
If you have used the EMP connector codes as it is, please change it in a proper manner and change it license header.

Reply:
fixed